operator/src/main/java/org/bf2/operator/operands/DeploymentOperand.java

```diff
@@ -0,0 +1,58 @@
+package org.bf2.operator.operands;
+
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import org.bf2.common.OperandUtils;
+import org.bf2.operator.resources.v1alpha1.ManagedKafka;
+import org.bf2.operator.resources.v1alpha1.ManagedKafkaCondition.Status;
+
+import javax.inject.Inject;
+
+import java.util.Objects;
+
+public abstract class DeploymentOperand implements Operand<ManagedKafka> {
+
+    @Inject
+    protected KubernetesClient kubernetesClient;
+
+    @Inject
+    protected AbstractKafkaCluster kafkaCluster;
+
+    protected abstract Deployment cachedDeployment(ManagedKafka managedKafka);
+
+    protected void createOrUpdate(Deployment deployment) {
+        OperandUtils.createOrUpdate(kubernetesClient.apps().deployments(), deployment);
+    }
+
+    protected void createOrUpdate(Service service) {
+        OperandUtils.createOrUpdate(kubernetesClient.services(), service);
+    }
+
+    public abstract Deployment deploymentFrom(ManagedKafka managedKafka, Deployment current);
```

Comment:
It might be slightly tidier to have this placed alongside the other declared abstract method on L22. WDYT?